### PR TITLE
added approve

### PR DESCRIPTION
**Pull request by Violent-Idiot**

### CODE REVIEW

File: test/sample-test.js [hunk 1]
```diff
@@ -38,9 +38,12 @@ describe("Aio Contract", function () {
...
       // const
       console.log(bal, contractAddr.address);
+      // daiContract
+      //   .transfer(contractAddr.address, ethers.utils.parseUnits("1", 0))
+      //   // .then((res) => console.log(res))
+      //   .catch((err) => console.log(err));
       daiContract
-        .transfer(contractAddr.address, ethers.utils.parseUnits("1", 0))
-        // .then((res) => console.log(res))
+        .approve(contractAddr.address, 1)
```
Comment (pradyuman-verma) [inline test/sample-test.js]: The amount is too low.
Use ethersjs for amounts.
To pass 1 DAI:
```js
ethers.utils.parseUnits("1", 18);
```

File: test/sample-test.js [hunk 2]
```diff
@@ -60,13 +63,10 @@ describe("Aio Contract", function () {
       //   });
       // .then((res) => console.log(res))
       // .catch((err) => console.log(err));
-      await aioToken.connect(owner).ERCdeposit(token, 1, {
-        // from: owner.address,
-        // value: ethers.utils.parseUnits("1", 0),
-      });
+      await aioToken.connect(owner).ERCdeposit(token, 1);
```
Comment (pradyuman-verma) [inline test/sample-test.js]: Try using more than 1 Wei!
At least 1 DAI

File: test/sample-test.js [hunk 3]
```diff
@@ -38,9 +38,12 @@ describe("Aio Contract", function () {
       );
```
Comment (pradyuman-verma) [inline test/sample-test.js]: > const signer = new ethers.Wallet(process.env.SECRET_KEY, provider);

No need to use this

Comment (Violent-Idiot) [reply test/sample-test.js]: Error: sending a transaction requires a signer (operation="sendTransaction", code=UNSUPPORTED_OPERATION, version=contracts/5.5.0)

**Pull request merged**